eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -37,12 +37,14 @@ jobs:
         CMAKE_GENERATOR: 'Visual Studio 16 2019'
         CMAKE_GENERATOR_PLATFORM: Win32
         CmakeArgs: ' -DBUILD_TRANSPORT_CURL=ON' #Leaving curl on here to explicitly test what the default behavior is on windows.
+        BuildArgs: '-v --parallel 8'
```

Comment:
If we build on 32-core nodes then setting to 8 is only going to use 25% of the cores.  It would be more efficient to use the number of cpu threads as the parallel count

Reply:
I will create a new issue to get the number of logical processors and use it here.
The challenge is that this is yaml config file, so, CMake hasn't run at this point yet.
Actually, nothing has run. (not even a powershell script).

The yaml configuration needs to be updated to un something like `WMIC CPU Get NumberOfLogicalProcessors` with powershell and use the result to set this argument.

This requires some extra effort and engSys knowledge :) , So, I would wait on it for Daniel

Reply:
issue: https://github.com/Azure/azure-sdk-for-cpp/issues/1187